### PR TITLE
Fix AFL test in flambda2

### PR DESCRIPTION
**Pull request by ncik-roberts**
This PR fixes a test failure that's been around since #1196, when we added AFL support for flambda2. It's a more principled version of #1789, which you can refer to to see what test is failing.

We fix the test failure by preventing a call to `CamlinternalLazy.force` from being inlined by flambda2 when `-afl-instrument` is in force. It's desirable to avoid this inlining because:
  * If the code is inlined, then AFL instrumentation will be inserted in all the branches.
  * `CamlinternalLazy.force` branches on whether the object header is `Forward_tag`, `Lazy_tag`, or something else.
  * When a lazy object with `Forward_tag` is traversed by the GC, the lazy object will silently be replaced by its contents, changing its tag.

In short, the current state of the world violates @stedolan's manifesto from the original PR that tries to fix this problem, ocaml/ocaml#1754:
> code that does the same thing twice should produce exactly the same instrumentation output both times

and this violation makes the `laziness` test in `ocaml/testsuite/tests/afl-instrumentation/test.ml` fail in flambda2.

This is small change on top of ocaml/ocaml#1754, which is the original attempt to prevent this problem. Flambda2 would actually go ahead and inline this attempted fix, though, and closure/flambda would not &mdash; that's why we haven't run into this problem until now.

### CODE REVIEW

**Comment by ncik-roberts** [comment]
@stedolan &mdash; I'm also interested in hearing whether you agree that this should be upstreamed. Given that the intent of this translation is expressly to avoid inlining the call, it seems correct to fill in `Never_inlined` here.

**Comment by stedolan** [comment]
Ahah! Well spotted and thank you for the fix. I agree this should definitely be upstreamed - it's clearly a bug, regardless of whether upstream can currently hit it.

**Review by stedolan — APPROVED** [review]
